static/app/components/forms/selectControl.tsx

```diff
@@ -331,6 +331,7 @@ function SelectControl<OptionType extends GeneralSelectValue = GeneralSelectValu
     MultiValueRemove,
     LoadingIndicator: SelectLoadingIndicator,
     IndicatorSeparator: null,
+    Option,
```

Comment:
We do override Option in a few places that may need some love.

Reply:
Yeah the plan is to refactor those custom components in future PRs once we commit this new, more flexible design first. Hopefully the new design (with optional descriptions, leading/trailing items,…) will account for the majority of those cases.